server/src/integration/java/org/opentosca/toscana/plugins/kubernetes/model/KubernetesPushingGopherIT.java

```diff
@@ -0,0 +1,144 @@
+import okhttp3.Request;
+import okhttp3.Response;
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.Collections.singletonMap;
+import static org.mockito.Mockito.mock;
+import static org.opentosca.toscana.plugins.kubernetes.KubernetesPlugin.DOCKER_PUSH_TO_REGISTRY_PROPERTY_KEY;
+import static org.opentosca.toscana.plugins.kubernetes.KubernetesPlugin.DOCKER_REGISTRY_PASSWORD_PROPERTY_KEY;
+import static org.opentosca.toscana.plugins.kubernetes.KubernetesPlugin.DOCKER_REGISTRY_REPOSITORY_PROPERTY_KEY;
+import static org.opentosca.toscana.plugins.kubernetes.KubernetesPlugin.DOCKER_REGISTRY_URL_PROPERTY_KEY;
+import static org.opentosca.toscana.plugins.kubernetes.KubernetesPlugin.DOCKER_REGISTRY_USERNAME_PROPERTY_KEY;
+
+public class KubernetesPushingGopherIT extends KubernetesLampIT {
+
+    private static final Logger logger = LoggerFactory.getLogger(KubernetesPushingGopherIT.class);
+
+    private static final String EXPECTED_TAG = "127.0.0.1:5000/test/gopher:app";
+
+    private List<String> runningContainers = new ArrayList<>();
+
+    public KubernetesPushingGopherIT() throws Exception {
+        DockerClient client = DefaultDockerClient.fromEnv().build();
+        logger.info("Downloading registry image");
+        client.pull("registry:2");
+
+        final Map<String, List<PortBinding>> ports = singletonMap(
+            "5000/tcp", Collections.singletonList(PortBinding.of("0.0.0.0", 5000)));
+        final HostConfig hostConfig = HostConfig.builder().portBindings(ports)
+            .build();
+
+        logger.info("Creating Local Registry Container");
+        ContainerConfig config = ContainerConfig.builder()
+            .hostConfig(hostConfig)
+            .image("registry:2").build();
+        String id = client.createContainer(config).id();
+        logger.info("Registry container id: {}", id);
+
+        logger.info("Starting registry container");
+        client.startContainer(id);
+        this.runningContainers.add(id);
+    }
+
+    @Override
+    protected void onSuccess(File outputDir) throws Exception {
+        DockerClient client = DefaultDockerClient.fromEnv().build();
+        client.pull(EXPECTED_TAG);
+
+        Map<String, List<PortBinding>> ports = singletonMap(
+            "8080/tcp", Collections.singletonList(PortBinding.of("0.0.0.0", 17717)));
+        HostConfig hostConfig = HostConfig.builder().portBindings(ports)
+            .build();
+
+        ContainerConfig config = ContainerConfig.builder()
+            .hostConfig(hostConfig)
+            .image(EXPECTED_TAG).build();
+        String id = client.createContainer(config).id();
+        this.runningContainers.add(id);
+
+        client.startContainer(id);
+
+        Thread.sleep(5000);
+        OkHttpClient httpClient = new OkHttpClient();
+        Request request = new Request.Builder().url("http://127.0.0.1:17717/").get().build();
```

Comment:
Consider to make the port `17717` as a constant. I already saw this port above

Reply:
Had to make the port dynamic anyways.